ciao-controller/main.go

```diff
@@ -268,12 +289,27 @@ func (c *controller) startCiaoService() error {
 	})
 
 	if err != nil {
-		return err
+		return nil, err
 	}
 
 	service := fmt.Sprintf(":%d", controllerAPIPort)
 
-	glog.Infof("Starting ciao API on port %d\n", controllerAPIPort)
+	server := &http.Server{
+		Handler: r,
+		Addr:    service,
+	}
+
+	return server, nil
+}
 
-	return http.ListenAndServeTLS(service, httpsCAcert, httpsKey, r)
+func (c *controller) ShutdownHTTPServers() {
+	glog.Warning("Shutting down HTTP servers")
+	for _, server := range c.httpServers {
+		ctx, cancel := context.WithTimeout(context.Background(), time.Second*5)
```

Comment:
Maybe we should call shutdown of each server in parallel.  Otherwise, we could end up waiting up to 15 seconds for controller to shutdown in the worst case.